react-native/react/login2/register/set-public-name/index.js

```diff
@@ -0,0 +1,50 @@
+'use strict'
+/* @flow */
+
+import React, {Component} from '../../../base-react'
+import {setDeviceName} from '../../../actions/login2'
+import Render from './index.render'
+
+export default class SetPublicName extends Component {
+  nameTaken (deviceName) {
+    return this.props.existingDevices && this.props.existingDevices.indexOf(deviceName) !== -1
+  }
+
+  submitEnabled (deviceName) {
+    return deviceName && deviceName.length && !this.nameTaken(deviceName)
+  }
+
+  render () {
+    return (
+      <Render
+        onSubmit={ deviceName => this.props.dispatch(setDeviceName(deviceName)) }
+        nameTaken={ deviceName => this.nameTaken(deviceName) }
+        submitEnabled={ deviceName => this.submitEnabled(deviceName) }
```

Comment:
No, I was thinking that the `Render` component here could work exactly like [`TextInput`](https://facebook.github.io/react-native/docs/textinput.html#content):

```jsx
<Render
  onChangeDeviceName={ deviceName => this.setState({ deviceName }) }
  onSubmit={ () => this.props.dispatch(setDeviceName(this.state.deviceName)) }
  nameTaken={this.nameTaken()}
  submitEnabled={this.submitEnabled()}
  {...this.props}
/>
```

The `Render` component wouldn't have state. It would call `onChangeDeviceName` when it got an updated value, and `nameTaken` and `submitEnabled` would both be value props instead of functions.